graylog2-server/src/main/java/org/graylog2/bootstrap/preflight/GraylogPreflightGeneratePeriodical.java

```diff
@@ -16,71 +16,137 @@
         this.nodePreflightConfigService = nodePreflightConfigService;
         this.csrStorage = csrStorage;
         this.certMongoStorage = certMongoStorage;
+        this.caService = caService;
+        this.passwordSecret = passwordSecret;
+        this.configuration = configuration;
+        this.nodeService = nodeService;
+        this.okHttpClient = okHttpClient;
+        buildTempHttpClient();
+    }
+
+    // building a non checking httpclient
+    private void buildTempHttpClient() {
+        try {
+            TrustManager[] trustAllCerts = new TrustManager[]{
+                    new X509TrustManager() {
+                        @Override
+                        public void checkClientTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+                        }
+
+                        @Override
+                        public void checkServerTrusted(java.security.cert.X509Certificate[] chain, String authType) {
+                        }
+
+                        @Override
+                        public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                            return new java.security.cert.X509Certificate[]{};
+                        }
+                    }
+            };
+
+            SSLContext sslContext = SSLContext.getInstance("SSL");
+            sslContext.init(null, trustAllCerts, new java.security.SecureRandom());
+
+            OkHttpClient.Builder newBuilder = new OkHttpClient.Builder();
+            newBuilder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) trustAllCerts[0]);
+            newBuilder.hostnameVerifier((hostname, session) -> true);
+            newBuilder.authenticator(new Authenticator() {
+                public Request authenticate(Route route, Response response) throws IOException {
+                    String credential = Credentials.basic("admin", "admin");
+                    return response.request().newBuilder().header("Authorization", credential).build();
+                }
+            });
+
+            tmpClient = newBuilder.build();
+        } catch (Exception ex) {
+            LOG.error("Could not create tmp okhttpclient " + ex.getMessage(), ex);
+        }
     }
 
     @Override
     public void doRun() {
         LOG.debug("checking if there are configuration steps to take care of");
-        final Path caKeystorePath = Path.of(caKeystoreFilename);
-
-        if(!Files.exists(caKeystorePath)) {
-            LOG.warn("mandatory keystore file does not exist.");
-            return;
-        }
 
         try {
-            char[] password = DEFAULT_PASSWORD.toCharArray();
-            KeyStore caKeystore = KeyStore.getInstance("PKCS12");
-            caKeystore.load(new FileInputStream(caKeystorePath.toFile()), password);
-
+            final var password = configuration.getCaPassword() != null ? configuration.getCaPassword().toCharArray() : passwordSecret.toCharArray();
```

Comment:
There may be a bug.
caService.loadKeyStore() can load keystore from many locations.
If you have CaPassword set, and service method will try to load from Mongo, which is protected by secret, this one can simply fail.

Reply:
(MAJOR) I still see inconsistencies in line 139 here and in `configuredCaInConfExists` method of `CaService`. 
The check to decide which password to choose should be the same in both places!